src/Ting/UnitOfWork.php

```diff
@@ -58,6 +59,9 @@ public function __construct(
         $this->connectionPool     = $connectionPool;
         $this->metadataRepository = $metadataRepository;
         $this->queryFactory       = $queryFactory;
+        $this->entities = new WeakMap();
+        $this->entitiesChanged = new WeakMap();
```

Comment:
Thanks for the fast reply @xavierleune 

If my understanding is correct, it  means we should preserve `$entitiesShouldBePersisted` as an array

Reply:
@Ngob @xavierleune I've switch back  `$entitiesShouldBePersisted` to an array, however this may have a impact in performances.
Let me explain, `UnitofWork::process()` loops over the property `$entitiesShouldBePersisted` and forward the uuid to methods `process*()`. In order to retrieve the corresponding entity, I need to iterate on the WeakMap $entities.
I don't kwow about big of an impact this this for performance.

Any thoughts ?

Reply:
Actually you should keep the entity in `$entitiesShouldBePersisted`, if not we are back to the same issue. No `tingUuid` should exists anymore.
You can index `$entitiesShouldBePersisted` with `spl_oject_hash` if you need to have a unique identifier (note sure if needed). Sounds good ?

Reply:
Something bugles me, why change $entitiesShouldBePersisted to an array ? It only stores the persistence state. However, the entity is actually stored in `$entityChanged` (cf methods processManaged, processNew, processDelete).
Am I wrong ?